Add NSCameraUsageDescription to Info.plist

To use programs which access the camera such as [OpenCV](https://opencv.org/), the key [NSCameraUsageDescription](https://developer.apple.com/documentation/bundleresources/information_property_list/nscamerausagedescription) needs to be in the Info.plist. Adding the key enables Emacs to request camera permissions when a program tries to access the camera, instead of crashing the program.

Is there a way to add this functionality?

@vedkothavade thank you for script to test it. This is what I saw after running it:

<img width="260" alt="image" src="https://user-images.githubusercontent.com/6507913/204208723-53aae004-67fd-4dfc-9e60-2b75187f2925.png">

The script failed with the following error:

```
OpenCV: not authorized to capture video (status 0), requesting...
OpenCV: camera failed to properly initialize!
Traceback (most recent call last):
  File "/Users/d12frosted/python-open-cv-test/run.py", line 5, in <module>
    cv2.imshow('frame', frame)
cv2.error: OpenCV(4.6.0) /Users/xperience/actions-runner/_work/opencv-python/opencv-python/opencv/modules/highgui/src/window.cpp:967: error: (-215:Assertion failed) size.width>0 && size.height>0 in function 'imshow'
```

Running it for the second time resulted in a new window with me inside. Neat.

Seems like working 🎉

--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -192,6 +192,9 @@ class EmacsPlusAT29 < EmacsBase
       # inject PATH to Info.plist
       inject_path
 
+      # inject description for protected resources usage
+      inject_protected_resources_usage_desc
+
       # Replace the symlink with one that avoids starting Cocoa.
       (bin/"emacs").unlink # Kill the existing symlink
       (bin/"emacs").write <<~EOS

--- a/Library/EmacsBase.rb
+++ b/Library/EmacsBase.rb
@@ -80,4 +80,14 @@ class EmacsBase < Formula
       system "which", "ls"
     end
   end
+
+  def inject_protected_resources_usage_desc
+    ohai "Injecting description for protected resources usage"
+    app = "#{prefix}/Emacs.app"
+    plist = "#{app}/Contents/Info.plist"
+
+    system "/usr/libexec/PlistBuddy -c 'Add NSCameraUsageDescription string' '#{plist}'"
+    system "/usr/libexec/PlistBuddy -c 'Set NSCameraUsageDescription Emacs requires permission to access the Camera.' '#{plist}'"
+    system "touch '#{app}'"
+  end
 end

